temoa/model_checking/unit_checking/unit_propagator.py

```
@@ -0,0 +1,315 @@
+                    # Simple string parsing to extract denominator
+                    try:
+                        # Handle potential parentheses e.g. "Mdollar / (GW)"
+                        parts = units.split('/', 1)
+                        if len(parts) == 2:
+                            cap_unit = parts[1].strip()
+                            # Clean up parentheses
+                            cap_unit = cap_unit.replace('(', '').replace(')', '').strip()
+                            if cap_unit:
+                                result[tech] = cap_unit
+                    except Exception:
+                        pass
```

Ruff (0.14.8)

`try`-`except`-`pass` detected, consider logging the exception

(S110)